### PR TITLE
Convert deprecated AbstractDataStore usage to ContentDataStore

### PR DESCRIPTION
**Pull request by tbarsballe**
A test case in the netcdf-out module uses AbstractDataStore. This PR modifies this to use ContentDataStore instead.

### CODE REVIEW

**Comment by tbarsballe** [comment]
So, looks like:
a) These tests ought to run fine without netcdf-out native libs, as the Travis failures are the same ones I see on my machine (which makes sense)
b) The test failures _appear_ to be unrelated to the AbstractDataStore -> ContentDataStore change, which is ... odd.

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStore.java [hunk 1]
```diff
@@ -94,377 +83,304 @@
...
+                } catch (Exception e) {
+
+                    e.printStackTrace();
+                    throw new RuntimeException(e);
+                }
+            }
 
+            @Override
+            protected SimpleFeatureType buildFeatureType() throws IOException {
+                return null;
```
Comment (aaime) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStore.java]: Null? This method is the one that builds the feature type on behalf of the content store, shouldn't it return FEATURE_TYPE?

Comment (tbarsballe) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStore.java]: Whoops, missed that one when autogenerating methods. Definitely should return feature type here.

**Comment by tbarsballe** [comment]
The remaining two failures appear to be caused by a difference of a few milliseconds in the time bounds / filter. Looking into it...

**Comment by tbarsballe** [comment]
Date issues should be fixed. I'm going to wait and verify the Travis Build actually passes (works on my machine), then squash the excess commits together and merge.

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java [hunk 2]
```diff
@@ -5,24 +5,38 @@
...
 import java.io.Serializable;
+import java.util.Collections;
 import java.util.Map;
 
 import org.geotools.data.AbstractDataStoreFactory;
 import org.geotools.data.DataStore;
 import org.geotools.data.DataStoreFactorySpi;
 
-public class MultiDimDataStoreFactory extends AbstractDataStoreFactory {
+public class MultiDimDataStoreFactory implements DataStoreFactorySpi {
```
Comment (jodygarnett) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java]: I think AbstractDataStoreFactory still exists ...

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java [hunk 3]
```diff
@@ -35,4 +49,19 @@ public String getDescription() {
         return new DataStoreFactorySpi.Param[] { new Param("ParentLocation") };
     }
 
+    @Override
+    public boolean canProcess(Map<String, Serializable> map) {
+        File dir = new File((String) map.get("ParentLocation"));
+        return dir.exists();
```
Comment (jodygarnett) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java]: There is a utility method to change the map against parameter info you could use.

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java (hunk 2 above)
Comment (tbarsballe) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStoreFactory.java]: It does, but is deprecated.

**Pull request merged**